### PR TITLE
Clarifies restrictions on fields

### LINKED ISSUE #194: [PE-D][Tester B] Accepts Invalid Email Input
It is not possible to have an email address like "aikenduit@hotmail" without a valid top-level domain (TLD) like ".com," ".net," or ".org." An email address without a TLD is considered incomplete and would not work as a valid email address. In the case of "aikenduit@hotmail," it lacks the TLD and would not be a functional email address.

-------------
Labels: `severity.Low` `type.FunctionalityBug`
original: kimshitong/ped#3

### PR DESCRIPTION
**Pull request by xavierpok**
Closes issue #194  by clarifying it's intended behaviour

Also improves upon the UG by listing out restrictions for our fields.

(Email parsing is surprisingly complex so the section is a killer)

Closes #194

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -292,18 +292,18 @@ Format: `exit`
...
+| a/     | SCHEDULE_NAME | Alphanumeric characters and whitespace, non-empty.                                                                                                                                                                                                                                                | Only usable in `schedule` command                                                                                                                                      |
+| o/     | NOTE          | Alphanumeric characters, punctuation(any of ``!"#$%&'()*+,-./:;<=>?@[\]^`{\| }~``)  and whitespace. Maximum of 1000 characters.                                                                                                                                                                   | Only usable in `note` command                                                                                                                                          |
```
Comment (imkwokyong) [inline docs/UserGuide.md:307]: Can add here that it could be empty

Comment (imkwokyong) [reply docs/UserGuide.md:307]: Punctuation-wise no problem, checked in app and every punctuation here is accepted.

File: docs/UserGuide.md [hunk 2]
```diff
@@ -292,18 +292,18 @@ Format: `exit`
...
+| n/     | NAME          | Alphanumeric characters and whitespace, non-empty.                                                                                                                                                                                                                                                | -                                                                                                                                                                      |
+| p/     | PHONE_NUMBER  | 3-15 numeric characters.                                                                                                                                                                                                                                                                          | -                                                                                                                                                                      |
```
Comment (imkwokyong) [inline docs/UserGuide.md:301]: Suggest to add non-empty here too

**Review by imkwokyong — CHANGES_REQUESTED** [review]
Just some nits and need to resolve merge conflicts. Thank you for helping to update the documentation!

**Comment by xavierpok** [comment]
Merge conflicts resolved & changes implemented.

Of note :

`note` in fact allows empty fields
Added note that whitespaces are stripped.

**Review by imkwokyong — APPROVED** [review]
LGTM!

**Pull request merged**